Foundation/NSArray.swift

```diff
@@ -589,22 +624,42 @@ open class NSArray : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, NSCo
         
         return lastEqual ? result + 1 : result
     }
-    
+
+    public convenience init(contentsOf url: URL, error: ()) throws {
+        let plistDoc = try Data(contentsOf: url)
+        guard let plistArray = try PropertyListSerialization.propertyList(from: plistDoc, options: [], format: nil) as? Array<Any>
+            else {
+                throw NSError(domain: NSCocoaErrorDomain, code: CocoaError.propertyListReadCorrupt.rawValue, userInfo: [NSURLErrorKey : url])
+        }
+        self.init(array: plistArray)
+    }
+
+    @available(*, deprecated)
     public convenience init?(contentsOfFile path: String) {
-        self.init(contentsOfURL: URL(fileURLWithPath: path))
+        do {
+            try self.init(contentsOf: URL(fileURLWithPath: path), error: ())
+        } catch {
+            return nil
+        }
     }
-    
-    public convenience init?(contentsOfURL url: URL) {
+
+    @available(*, deprecated)
+    public convenience init?(contentsOf url: URL) {
         do {
-            guard let plistDoc = try? Data(contentsOf: url),
-            let plistArray = try PropertyListSerialization.propertyList(from: plistDoc, options: [], format: nil) as? Array<Any>
-             else { return nil }
-            self.init(array: plistArray)
+            try self.init(contentsOf: url, error: ())
         } catch {
             return nil
         }
     }
-    
+
+    open func pathsMatchingExtensions(_ filterTypes: [String]) -> [String] {
+        NSUnimplemented()
+    }
+
+    open func shuffled() -> [Any] {
```

Comment:
hmm where does this come from?

Reply:
https://developer.apple.com/documentation/foundation/nsarray/1640855-shuffled

Reply:
That comes from GameplayKit; which I am certain is not under the realms of what we are going to be porting to Linux ;)

Reply:
Surely only `func shuffled(using: GKRandomSource)` requires GameKit? A shuffled function might be useful although I can always make it unavailable if not. Im sure there are lots of example interview answers that can be used for an implementation :)

Reply:
There are a number of issues with claiming Foundation provides a shuffle - one case is performance, another case is cryptographic security, etc. I think we should not be adding new API to Foundation here and if a developer needs a shuffle they could probably be better serviced by adding a category on Collection and tailor it to their needs performance, cryptographic needs etc.

For now lets just omit this part of the change and address any additional APIs separate from this bug fix